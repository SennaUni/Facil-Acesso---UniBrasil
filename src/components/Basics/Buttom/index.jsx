import React from 'react';
import { ActivityIndicator } from 'react-native';

import { Container, Title } from './styles';

export function Buttom({ title, loading = false, ...rest }) {
  return (
    <>
      { loading ? (
        <Container {...rest} loading={true} disabled={true}>
          <ActivityIndicator
            size='small'
            color='#FFF'
          />
          <Title>Loading</Title>
        </Container>
      ) : (
        <Container {...rest} >
          <Title>{title}</Title>
        </Container>
      )}
    </>
  );
}