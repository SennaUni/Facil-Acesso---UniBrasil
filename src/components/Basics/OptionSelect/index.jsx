import React from 'react';

import { FontAwesome } from '@expo/vector-icons';

import { Container, Picture, Label } from './styles';

export function OptionSelect({ item, selectedValue, onPress }) {

  return (
    <Container 
      backGround={item.id === selectedValue?.id ? '#EEE' : '#FFF'}
      onPress={onPress}
    >
      <Picture 
        source={{
          uri: item.avatarUrl,
        }}
      />
      <Label 
        fontWeight={item.id === selectedValue?.id ? 'bold' : 'normal'} 
      >
        {item.name}
      </Label>
      {/* { item.id === selectedValue?.id && (
        <FontAwesome
        name='check-square'
        size={22}
        color={'green'}
      />
      )} */}
    </Container>
  );
}
