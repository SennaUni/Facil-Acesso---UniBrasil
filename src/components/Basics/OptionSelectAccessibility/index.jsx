import React from 'react';

import { FontAwesome } from '@expo/vector-icons';

import { Container, Picture, Label } from './styles';

export function OptionSelectAccessibility({ item, selectedValue, onPress }) {

  return (
    <Container 
      backGround={item.id === selectedValue?.id ? '#8241B8' : '#FFF'}
      onPress={onPress}
    >
      <Picture 
        source={{
          uri: item.avatarUrl,
        }}
      />
      <Label 
        color={item.id === selectedValue?.id ? '#FFF' : '#555'}
      >
        {item.name}
      </Label>
      {/* { item.id === selectedValue?.id && (
        <FontAwesome
        name='check-square'
        size={22}
        color={'green'}
      />
      )} */}
    </Container>
  );
}
