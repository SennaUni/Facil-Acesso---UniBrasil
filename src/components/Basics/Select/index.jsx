import React, { useState, useRef, useEffect } from 'react';

import { TouchableOpacity, Modal, FlatList, SafeAreaView } from 'react-native';

import { FontAwesome } from '@expo/vector-icons';

import { useField } from '@unform/core';

import { Label, Container, ComboBox, ModalHeader, SelectTitle, ModalTitle, ModalClose } from './styles';

export function Select({ name, options, text, label, OptionComponent }) {
  const selectRef = useRef(null)

  // const { fieldName, registerField } = useField('teste');

  const [openModal, setOpenModal] = useState(false);
  const [selectedValue, setSelectedValue] = useState();

  function renderOption(item) {
    return (
      <OptionComponent 
        item={item}
        selectedValue={selectedValue}
        onPress={() => {
          setSelectedValue(item);
          setOpenModal(false);
        }}
      />
    )
  }

  // useEffect(() => {
  //   registerField({
  //     name: fieldName,
  //     ref: selectRef.current,
  //     path: 'value',
  //     getValue: ref => {
  //       return ref.value || ''
  //     },
  //     setValue: (ref, value) => {
  //       ref.setNativeProps({ text: value });
  //       ref.value = value
  //     },
  //     clearValue: ref => {
  //       ref.value = ''
  //     },
  //   })
  // }, [fieldName, registerField])

  return (
    <>
      <Label>
        {label}
      </Label>
      <Container >
        <ComboBox
          onPress={() => setOpenModal(true)}
        >
          <SelectTitle 
            numberOfLines={1}
          >
            {selectedValue?.name || text}
          </SelectTitle>
          <FontAwesome
            name='angle-down'
            size={26}
            color={'#000'}
          />
        </ComboBox>
        <Modal 
          animationType='slide'
          visible={openModal}
          onRequestClose={() => setOpenModal(false)}
        >
          <SafeAreaView>
            <ModalHeader>
              <TouchableOpacity
                onPress={() => setOpenModal(false)}
              >
                <FontAwesome
                  name='angle-down'
                  size={30}
                  color={'#000'}
                />
              </TouchableOpacity>
              <ModalTitle>
                {text}
              </ModalTitle>
              <TouchableOpacity
                onPress={() => setOpenModal(false)}
              >
                <ModalClose>
                  Cancelar
                </ModalClose>
              </TouchableOpacity>
            </ModalHeader>
            <FlatList 
              data={options ?? []}
              keyExtractor={item => String(item.id)}
              renderItem={({ item }) => renderOption(item) }
            />
          </SafeAreaView>
        </Modal>
      </Container>
    </>
  );
}