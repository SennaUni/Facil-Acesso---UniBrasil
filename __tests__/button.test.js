import React from 'react';

import { create, act } from 'react-test-renderer';

import { Buttom } from '../src/components/Basics/Buttom';

describe('render correctly', () => {

  it('loading false', () => {
    const tree = create(
      <Buttom 
        title='TESTE'
        loading={true}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('loading true', () => {
    const tree = create(
      <Buttom 
        title='TESTE'
        loading={false}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});