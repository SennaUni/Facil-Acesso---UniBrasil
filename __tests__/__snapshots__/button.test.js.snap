// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly loading false 1`] = `
<BVLinearGradient
  colors={
    [
      4285281187,
      4286726584,
    ]
  }
  endPoint={
    {
      "x": 0.5,
      "y": 1,
    }
  }
  locations={null}
  startPoint={
    {
      "x": 0.5,
      "y": 0,
    }
  }
  style={
    [
      {
        "borderRadius": 25,
        "marginBottom": 20,
        "marginLeft": 10,
        "marginRight": 10,
        "marginTop": 20,
      },
    ]
  }
>
  <View
    accessibilityState={
      {
        "disabled": true,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "borderRadius": 20,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginBottom": 8,
        "marginTop": 10,
        "opacity": 0.6,
        "paddingBottom": 10,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 10,
      }
    }
  >
    <ActivityIndicator
      color="#FFF"
      size="small"
    />
    <Text
      style={
        [
          {
            "color": "#FFFFFF",
            "fontSize": 16,
            "fontWeight": "bold",
            "marginBottom": 0,
            "marginLeft": 8,
            "marginRight": 8,
            "marginTop": 0,
          },
        ]
      }
    >
      Loading
    </Text>
  </View>
</BVLinearGradient>
`;

exports[`render correctly loading true 1`] = `
<BVLinearGradient
  colors={
    [
      4285281187,
      4286726584,
    ]
  }
  endPoint={
    {
      "x": 0.5,
      "y": 1,
    }
  }
  locations={null}
  startPoint={
    {
      "x": 0.5,
      "y": 0,
    }
  }
  style={
    [
      {
        "borderRadius": 25,
        "marginBottom": 20,
        "marginLeft": 10,
        "marginRight": 10,
        "marginTop": 20,
      },
    ]
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "borderRadius": 20,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginBottom": 8,
        "marginTop": 10,
        "opacity": 1,
        "paddingBottom": 10,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 10,
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#FFFFFF",
            "fontSize": 16,
            "fontWeight": "bold",
            "marginBottom": 0,
            "marginLeft": 8,
            "marginRight": 8,
            "marginTop": 0,
          },
        ]
      }
    >
      TESTE
    </Text>
  </View>
</BVLinearGradient>
`;
